### ISSUE TITLE
core: refactor http webclient to merge all implementations into one

### ISSUE THREAD

**Issue by ngosang** [Jackett/Jackett#8529]
Currently we have 4 implementations of http webclient that should be merged into one.
Development in a new branch => https://github.com/Jackett/Jackett/tree/feature/web-client

**Comment by ngosang** [comment]
@cadatoiva All work to date is merged into master. There are some remaining work to merge both httpclient implementations into one, or better, remove `httpclient (v1)` since it's used only by Mono version and I think v2 is working too in Mono.

**Comment by garfield69** [comment]
this has broken libble, arenabg, horrible subs and liorencili on my windows 0.16.1260 going backto 0.16.1252 restores function.

**Comment by ngosang** [comment]
@garfield69 @ilike2burnthing I fixed #9569 and #9568
It's merged into master. This time I tested it in Mono 5.x too and I hope it does not break other things.

**Comment by ilike2burnthing** [comment]
Confirmed v0.16.1296.0 working on Windows for Horrible Subs and LiaorenCili, with and without proxy/VPN.

I take it that it's intentional/expected that we now just get the size of the returned data when using enhanced logging, rather than the actual HTML? e.g.
```
Debug WebClient(HttpWebClient).GetResultAsync(Url:http://www.cilinb1.xyz/list/1/MLjYAyMAO0O0OO0O0O/1.html) 
Debug WebClient(HttpWebClient): Returning OK => 15746 bytes 
Info Found 20 releases from LIAORENCILI
```

**Comment by garfield69** [comment]
it depends on the indexer, especially those written in C#.

**Comment by garfield69** [comment]
but if the previous version when you tested horriblesub produced debug html and this version does not then that is an oversight that should be corrected. @ngosang ?

**Comment by ilike2burnthing** [comment]
No, v0.16.1260 just gave the size, I didn't ask at the time because I wasn't sure if it was as a result of the same issue.

**Comment by ngosang** [comment]
I'm going to test. I think HTML is better than the size.

**Comment by ilike2burnthing** [comment]
Would it be anything to do with stringResult no longer being returned in WebClient.cs? https://github.com/Jackett/Jackett/commit/ac0ba0f9b11eeb5e42920d3eece222077ab90183#diff-ce6924490a23a706923db18e991d4ac1

**Comment by ngosang** [comment]
Fixed in #9589
That implementation improves previous code by adding Method (Get/Post) and printing HTML code in redirect requests too.

**Comment by ngosang** [comment]
I think the current implementation can be improved but it's good enough.
